examples/reshare/src/orchestrator/actor.rs

```diff
@@ -164,51 +175,107 @@ where
 
         // Wait for instructions to transition epochs.
         let mut engines = BTreeMap::new();
-        while let Some(transition) = self.mailbox.next().await {
-            match transition {
-                Message::Enter(transition) => {
-                    // If the epoch is already in the map, ignore.
-                    if engines.contains_key(&transition.epoch) {
+        loop {
+            select! {
+                message = pending_backup.next() => {
+                    // If a message is received in an unregistered sub-channel in the recovered or pending network,
+                    // attempt to forward the boundary finalization for the epoch.
+
+                    let Some((epoch, (from, _))) = message else {
+                        warn!("recovered/pending mux backup channel closed, shutting down orchestrator");
+                        break;
+                    };
+                    let Some(latest_epoch) = engines.keys().last().copied() else {
+                        debug!(epoch, ?from, "received message from unregistered epoch with no known epochs");
+                        continue;
+                    };
+                    if epoch as u64 >= latest_epoch {
+                        // The sender is operating in a newer epoch or the current, ignore. We cannot block them,
+                        // since they may be validly ahead.
+                        debug!(epoch, ?from, "received backup message from current or future epoch");
                         continue;
                     }
 
-                    // Enter the new epoch.
-                    let engine = self
-                        .enter_epoch(
-                            transition.epoch,
-                            transition.poly,
-                            transition.share,
-                            transition.participants,
-                            &mut pending_mux,
-                            &mut recovered_mux,
-                            &mut resolver_mux,
-                        )
-                        .await;
-                    engines.insert(transition.epoch, engine);
+                    // Fetch the finalization certificate for the last block within the subchannel's epoch.
+                    // If the node is state synced, marshal may not have the finalization locally, and the
+                    // peer will need to fetch it from another node on the network.
+                    let boundary_height = last_block_in_epoch(BLOCKS_PER_EPOCH, epoch as u64);
```

Comment:
I wonder if we should change the channel to u64? Would allow us to move away from the casting to u32 we below and casting up here?

I don't think we should ever really hit this but if we are using a varint its probably fine.